### PR TITLE
snapshot/delete: Implement snapshot delete all

### PR DESCRIPTION
**Pull request by rafikc30**
Glustercli will have an added feature of deleting all
snapshots in the cluster or all snapshots of a particular
volume

Signed-off-by: Mohammed Rafi KC <rkavunga@redhat.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
@rafikc30 can you add example CLI commands for this PR.

File: glustercli/cmd/snapshot-delete.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package cmd
 import (
 	"fmt"
 
+	"github.com/gluster/glusterd2/pkg/api"
```
Comment (Madhu-1) [inline glustercli/cmd/snapshot-delete.go]: add a new line here

File: glustercli/cmd/snapshot-delete.go [hunk 2]
```diff
@@ -19,24 +20,78 @@ var (
...
+}
+
+func snapshotDeleteAllCmdRun(cmd *cobra.Command, args []string) {
+	var snaps api.SnapListResp
+	var err error
+	volname := ""
+	if len(args) > 0 {
+		volname = args[0]
+	}
+
```
Comment (rishubhjain) [inline glustercli/cmd/snapshot-delete.go]: @rafikc30 I think there is a need to throw an error if the volume is not provided.

Comment (rafikc30) [reply glustercli/cmd/snapshot-delete.go]: Volume is an optional argument. If omitted, then perform the deletion of all snapshots in the cluster. I have updated the example cli commands in commit message

File: glustercli/cmd/snapshot-delete.go [hunk 3]
```diff
@@ -19,24 +21,78 @@ var (
...
+		return err
 	}
 	fmt.Printf("%s Snapshot deleted successfully\n", snapname)
+	return nil
+}
+
+func snapshotDeleteCmdRun(cmd *cobra.Command, args []string) {
+
+	if len(args) != 1 {
+		//snapshotDeleteAllCmdRun will take care of else case.
```
Comment (Madhu-1) [inline glustercli/cmd/snapshot-delete.go]: @rafikc30 just a question, `snapshotDeleteAllCmdRun` will take care if `len(args) != 1` right?
if yes, should this comment need to be updated?

Comment (rafikc30) [reply glustercli/cmd/snapshot-delete.go]: yes, you are right

File: glustercli/cmd/snapshot-delete.go [hunk 4]
```diff
@@ -19,24 +21,78 @@ var (
...
-		failure("Snapshot delete failed", err, 1)
+		return err
 	}
 	fmt.Printf("%s Snapshot deleted successfully\n", snapname)
+	return nil
+}
+
+func snapshotDeleteCmdRun(cmd *cobra.Command, args []string) {
+
+	if len(args) != 1 {
```
[the remark this answers is not in the document]
Comment (rafikc30) [reply glustercli/cmd/snapshot-delete.go]: I thought the command `all` and 'voname' will also count in the handler snapshotDeleteCmdRun. I was wrong, I will remove the code and will make cobra.ExactArgs(1)

**Pull request merged**